prov/util/src/util_shm.c

```diff
@@ -145,6 +145,39 @@ size_t smr_calculate_size_offsets(size_t tx_count, size_t rx_count,
 	return total_size;
 }
 
+static int smr_retry_map(const char *name, int *fd)
+{
+	char tmp[NAME_MAX];
+	struct smr_region *old_shm;
+	struct stat sts;
+	int shm_pid;
+
+	*fd = shm_open(name, O_RDWR | O_CREAT, S_IRUSR | S_IWUSR);
+	if (*fd < 0)
+		return -errno;
+
+	old_shm = mmap(NULL, sizeof(*old_shm), PROT_READ | PROT_WRITE,
+		       MAP_SHARED, *fd, 0);
+	if (old_shm == MAP_FAILED)
+		return -errno;
+
+	if (old_shm->version > SMR_VERSION) {
+		munmap(old_shm, sizeof(*old_shm));
+		return -FI_EBUSY;
+	}
+	shm_pid = old_shm->pid;
+	munmap(old_shm, sizeof(*old_shm));
+
+	if (!shm_pid)
+		return FI_SUCCESS;
+
+	memset(tmp, 0, sizeof(tmp));
+	snprintf(tmp, sizeof(tmp), "/proc/%d", shm_pid);
+
+	return stat(tmp, &sts) == -1 && errno == ENOENT ?
+		FI_SUCCESS : -FI_EBUSY;
+}
```

Comment:
I think we need to close and unlink on error

Reply:
Good point. Updated with error path that also closes the fd on error.